core/server/common/src/main/java/alluxio/StorageTierAssocImplementation.java

```diff
@@ -28,7 +28,8 @@
  * {@link ServerConfiguration}.
  */
 @ThreadSafe
-public abstract class AbstractStorageTierAssoc implements StorageTierAssoc {
+public final class StorageTierAssocImplementation
```

Comment:
nit: In the current code I don't see pattern like using interface `Foo` and implementation class `FooImplementation`. It's more frequent that we use `interface Foo` then `class DefaultFoo implements Foo`. I suggest keeping the naming consistent.

Taking a step back, if an interface only has one implementation and there's no obvious need to extend that, I think it's fair to just remove that interface and only keep the implementation class. I'm fine with that if that makes your naming easier.

Reply:
Ya, I hate the name `DefaultXXX`, makes no sense to me. I'd remove the interface if it's not in a separate package. Basically the interface is in common but the implementation is not. Eventually I'd like to restructure the packaging so we can remove the interface as well.